Hackney.Shared.CautionaryAlerts/Factories/ResponseFactory.cs

```diff
@@ -78,11 +78,18 @@ public static CautionaryAlertResponse ToResponse(this CautionaryAlertListItem do
                 Description = domain.CautionOnSystem,
                 Reason = domain.Reason,
                 AssureReference = domain.AssureReference,
-                PersonId = Guid.Parse(domain.PersonId),
+                PersonId = CheckIfExists(domain.PersonId),
                 PersonName = domain.Name
             };
         }
 
+        private static Guid? CheckIfExists(string personId)
```

Comment:
The name of the method is misleading. The "CheckIfExists" suggests that the answer will be returned in a `Boolean` form, however, it isn't.

I recommend either:
1. refactoring this method into 2 based on their 2 separate responsibilities (validate & parse), or ...
2. rename the method into something like `ParseGuidIfExists`.